code/modules/projectiles/guns/projectile/CMturret.dm

```diff
@@ -0,0 +1,223 @@
+/mob/living/carbon/human
+	var/obj/structure/machinegun/mounted
+
+/mob/living/carbon/human/ClickOn(atom/A, params)
+    if(mounted && Adjacent(mounted))
+        if(next_move < world.time)
+            mounted.shoot(get_turf(A))
+    else
+        . = ..()
+
+
+/obj/structure/machinegun
+	name = "machine gun"
+	desc = "A stationary machine gun."
+	icon = 'icons/obj/structures.dmi'
+	icon_state = "mgun+barrier"
+	var/fire_sound = 'sound/weapons/Gunshot_smg.ogg'
+	var/empty_sound = 'sound/weapons/empty.ogg'
+	var/ammo_type = "/obj/item/projectile/bullet/a762"
+	var/ammo = 500
+	var/ammomax = 500
+	var/list/row1 = list()
+	var/list/row2 = list()
+	var/list/row3 = list()
+	var/mob/living/carbon/human/user
+	var/FIRETIME = 1 //tenths of seconds
+	density = 1
+	anchored = 0
+	flags = ON_BORDER
+	var/health = 500
+
+/obj/structure/machinegun/process()
+	var/mob/living/carbon/human/U
+	for(var/mob/living/carbon/human/H in range(0, src))
+		if(H)
+			U = H
+	if(U)
+		if(user && user != U)
+			user.mounted = null
+			user = null
+		user = U
+		user.mounted = src
+	else
+		if(user)
+			user.mounted = null
+			user = null
+
+/obj/structure/machinegun/New()
+	..()
+	processing_objects.Add(src)
+
+/obj/structure/machinegun/proc/shoot(var/turf/T)
+	if(ammo <= 0)
+		if(user)
+			playsound(src, empty_sound, 70, 1)
+			to_chat(user, "This machine gun is out of ammo!")
+
+		return
+	if(T && user && !user.incapacitated())
+		var/row = 0
+		if(row1.Find(T))
+			row = 1
+		else if(row2.Find(T))
+			row = 2
+		else if(row3.Find(T))
+			row = 3
+		if(row)
+			var/turf/shootfrom
+			switch(row)
+				if(1)
+					shootfrom = get_step(src, turn(dir, 90))
+				if(2)
+					shootfrom = get_step(src, dir)
+				if(3)
+					shootfrom = get_step(src, turn(dir, 270))
+			if(shootfrom)
+
+				var/turf/curloc = get_turf(shootfrom)
+				var/turf/targloc
+				switch(row)
+					if(1)
+						targloc = row1[7]
+					if(2)
+						targloc = row2[7]
+					if(3)
+						targloc = row3[7]
+				if (!istype(targloc) || !istype(curloc))
+					return
+				playsound(src, fire_sound, 50, 1)
+				var/obj/item/projectile/bullet/a762/bullet = new ammo_type(shootfrom)
```

Comment:
This should only specify type down to `/obj/item/projectile` since you can change the ammo_type variable to something that's not `/bullet/a762/`. Also I can't quite remember if the call to `new` will work properly, since `ammo_type` is text, not a path.